### PR TITLE
Upgrade mypy to 0.991

### PR DESCRIPTION
**Pull request by ashb**
Most of the changes are related to with `x = None`

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/cloud_sql.py [hunk 1]
```diff
@@ -686,8 +686,6 @@ class CloudSQLDatabaseHook(BaseHook):
     conn_type = "gcpcloudsqldb"
     hook_name = "Google Cloud SQL Database"
 
-    _conn = None
```
Comment (ashb) [inline airflow/providers/google/cloud/hooks/cloud_sql.py:694]: This field was never used in this hook.

File: airflow/stats.py [hunk 2]
```diff
@@ -342,7 +343,7 @@ def timer(self, stat=None, *args, tags=None, **kwargs):
 
 
 class _Stats(type):
-    factory = None
+    factory: Callable[[], StatsLogger]
```
Comment (ashb) [inline airflow/stats.py:364]: By not setting this to `None` we have easier typchecking of calling `cls.__class__.factory`

File: airflow/timetables/simple.py [hunk 3]
```diff
@@ -55,6 +55,9 @@ def serialize(self) -> dict[str, Any]:
     def infer_manual_data_interval(self, *, run_after: DateTime) -> DataInterval:
         return DataInterval.exact(run_after)
 
+    def validate(self):
```
Comment (ashb) [inline airflow/timetables/simple.py:58]: I'm not quite sure how these time tables were working. Mypy complained about creating an abstract class anyway (i guess it's a difference in what it thinks Protocols do vs how they work at runtime?)

PTAL at this change @uranusjr

File: airflow/www/extensions/init_appbuilder.py [hunk 4]
```diff
@@ -14,6 +14,8 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+
+# mypy: disable-error-code=var-annotated
```
Comment (ashb) [inline airflow/www/extensions/init_appbuilder.py:18]: This is to save making a lot of changes in this file.

File: tests/www/views/test_views_connection.py [hunk 5]
```diff
@@ -40,11 +41,10 @@
     "password": "admin",
 }
 
-CONNECTION_WITH_EXTRA = CONNECTION.update(
```
Comment (ashb) [inline tests/www/views/test_views_connection.py]: @ephraimbuddy I think this was your PR where we added it? Turns out that `CONNECTION_WITH_EXTRA` was None (update returns in place, it doesn't return.) Could you expand the test to check that x_secret and y_secret are actually masked please? (separate PR)

File: airflow/example_dags/plugins/workday.py [hunk 6]
```diff
@@ -78,6 +78,9 @@ def next_dagrun_info(
 
     # [END howto_timetable_next_dagrun_info]
 
+    def validate(self):
```
Comment (potiuk) [inline airflow/example_dags/plugins/workday.py:81]: should it be separated ? seems like "real" bugfix which might be hidden in big mypy change ?

Comment (potiuk) [reply airflow/example_dags/plugins/workday.py:81]: Scrap that. I know. comment below.

File: airflow/timetables/simple.py (hunk 3 above)
Comment (potiuk) [reply airflow/timetables/simple.py:58]: TIL.  Empty methods - whether ... or just with docstring **just work**. No matter if they are part of protocol or not:

This one works fine and exit with 0 error code:

```python
from typing_extensions import Protocol


class A(Protocol):
    def method(self) -> None:
        """Empty method"""

    def method2(self) -> None:
        ...


class B(A):
    def method3(self) -> None:
        ...


B().method()
B().method2()
B().method3()
```

Comment (ashb) [reply airflow/timetables/simple.py:58]: I suspect we are "unusual" in inheriting from a Protocol (we are treating it somewhat like an ABC) and I guess mypy is somehow ignoring the methods on here.

File: airflow/dag_processing/processor.py [hunk 7]
```diff
@@ -56,6 +56,8 @@
 from airflow.utils.state import State
 
 if TYPE_CHECKING:
+    import multiprocessing.context
+
```
Comment (uranusjr) [inline airflow/dag_processing/processor.py]:
```suggestion
```

File: airflow/example_dags/plugins/workday.py [hunk 8]
```diff
@@ -78,6 +78,9 @@ def next_dagrun_info(
...
+        pass
```
Comment (uranusjr) [inline airflow/example_dags/plugins/workday.py:82]: Doesn’t the base class provide a default (empty) implementation?

Comment (ashb) [reply airflow/example_dags/plugins/workday.py:82]: I think it does, by new version of mypy doesn't seem to see the impl from the Protocol baseclass for some reason (arguably a bug in mypy)

Comment (uranusjr) [reply airflow/example_dags/plugins/workday.py:82]: (GitHub needs a 🤦 reaction)

File: airflow/timetables/simple.py (hunk 3 above)
Comment (uranusjr) [reply airflow/timetables/simple.py:58]: PEP 544 has examples about subclassing a base protocol and paragraphs talking specifically about default implementations, so those should not be special in any way. Definitely a bug in Mypy.

File: tests/www/views/test_views_connection.py (hunk 5 above)
Comment (ephraimbuddy) [reply tests/www/views/test_views_connection.py]: Thanks. PR here: https://github.com/apache/airflow/pull/29011

File: airflow/timetables/simple.py (hunk 3 above)
Comment (potiuk) [reply airflow/timetables/simple.py:58]: :eyes: instead of :man_facepalming:

File: tests/www/views/test_views_connection.py (hunk 5 above)
Comment (potiuk) [reply tests/www/views/test_views_connection.py]: merged.

**Review by potiuk — APPROVED** [review]
LGTM pending one suggestion from @uranusjr

File: airflow/timetables/simple.py (hunk 3 above)
Comment (uranusjr) [reply airflow/timetables/simple.py:58]: I kind of figured out what’s going on. Mypy seems to treat a function with `pass` as “empty” and ignore its presence. It’s a bug, but easier to work around (by replacing `pass` with `return` in the protocol).

**Pull request merged**